module-infrastructure/security/src/main/kotlin/site/yourevents/principal/AuthDetails.kt

```diff
@@ -0,0 +1,25 @@
+package site.yourevents.principal
+
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.userdetails.UserDetails
+import java.util.Collections
+import java.util.UUID
+
+class AuthDetails(
+    private val uuid: UUID,
+    private val socialId: String,
+    private val role: String,
+) : UserDetails {
+    override fun getAuthorities(): Collection<GrantedAuthority?>? {
```

Comment:
이 부분에서 ?이 왜 사용된건지 알 수 있을까요?
저렇게 구현이 되면 컬렉션 자체도 null이 될 수 있고, 컬렉션 안의 GrantedAuthority도 null 될 수 있다는 것을 의미합니다. 그런데 getAuthorities()는 원칙적으로 null을 반환해서는 안된다고 알고 있습니다.
따라서 저의 생각으로는 ?를 제거하는 것이 맞다고 생각하는데 정완님은 어떻게 생각하시나요?

Reply:
아..너무 뼈아픈 지적입니다ㅠ.ㅠ 인텔리제이에서 자동으로 제공해주는 기능만 믿고 코드를 작성하면 이렇게 중요한 부분을 놓치게 되는 것 같습니다.ㅜㅜ

역시 승현님이 말씀해주신 대로 [공식 문서](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/userdetails/UserDetails.html)에서는 해당 메서드는 null을 절대 반환해서는 안된다고 설명하고 있습니다.

그리고 `GrantedAuthority)` 자체가 null인 경우는 의미가 없기에, 결국 nullable을 제거해주는 것이 맞다고 생각합니다.

다음과 같이 수정할 수 있을 것 같습니다!
```suggestion
    override fun getAuthorities(): Collection<GrantedAuthority> {
```

Reply:
확인하였습니다!!